src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Options/NuGetExternalSettingsProvider.cs

```diff
@@ -39,6 +40,11 @@ public virtual Task<ExternalSettingOperationResult<IReadOnlyList<EnumChoice>>> G
 
         internal virtual void VsSettings_SettingsChanged(object sender, EventArgs e)
         {
+            if (_suppressSettingValuesChanged)
```

Comment:
While writing package sources, we don't want the file watcher to raise `nuget.config` changed events. This suppression flag allows a page to temporarily disable raising change events to USX while it's responsible for writing to the nuget.config.